backend/api/auth.py

```
@@ -0,0 +1,297 @@
+from typing import Optional
+
+from models.database import get_db
+from models.models import User, APIKey, SubscriptionStatus
+
+router = APIRouter(prefix="/auth", tags=["authentication"])
+
+# Password hashing
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+# JWT settings (should be in environment variables in production)
+SECRET_KEY = "your-secret-key-change-in-production"  # TODO: Move to env var
```

Ruff (0.12.2)

Possible hardcoded password assigned to: "SECRET_KEY"

(S105)